tests/conftest.py

```diff
@@ -24,10 +24,10 @@
 from orion.core.worker.trial import Trial
 from orion.storage.base import get_storage, setup_storage, storage_factory
 from orion.storage.legacy import Legacy
-from orion.testing import OrionState, mocked_datetime
 
 # So that assert messages show up in tests defined outside testing suite.
 pytest.register_assert_rewrite("orion.testing")
+from orion.testing import OrionState, mocked_datetime
```

Comment:
Does the pytest.register_assert_rewrite if we import the module afterwards?

Reply:
I was getting a warning because it can't register assert rewrites if the module has already been imported. I think this is the correct ordering, but I might be wrong.

Reply:
Can you please mess with a test in orion.testing and see if the assert output is the verbose one from pytest or the basic one from python?

Reply:
With this ordering, we now get the verbose output:
```
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <test_asha.TestGenericASHA object at 0x7f1d6de97640>, config = {'num_brackets': 2, 'num_rungs': 5, 'repetitions': 3, 'seed': 123456}
space = None, kwargs = {}, base_algo_type = <class 'orion.algo.asha.ASHA'>
original_space = Space([Fidelity(name=f, low=1, high=10, base=2),
       Real(name=x, prior={uniform: (0, 1), {}}, shape=(), default value=None),
       Real(name=y, prior={uniform: (0, 1), {}}, shape=(), default value=None)])
algo = <orion.core.worker.primary_algo.SpaceTransformAlgoWrapper object at 0x7f1d6d9b0040>
@py_assert1 = <orion.algo.asha.ASHA object at 0x7f1d6dad96a0>, @py_assert3 = 200, @py_assert6 = 'BOB', @py_assert5 = False

    def create_algo(self, config=None, space=None, **kwargs):
        """Create the algorithm based on config.
    
        Parameters
        ----------
        config: dict, optional
            The configuration for the algorithm. ``self.config`` will be used
            if ``config`` is ``None``.
        space: ``orion.algo.space.Space``, optional
            Space object to pass to algo. The output of ``self.create_space()``
            will be used if ``space`` is ``None``.
        kwargs: dict
            Values to override algorithm configuration.
        """
        config = copy.deepcopy(config or self.config)
        config.update(kwargs)
        base_algo_type = orion.algo.base.algo_factory.get_class(self.algo_name)
        original_space = space or self.create_space()
        algo = create_algo(space=original_space, algo_type=base_algo_type, **config)
        algo.algorithm.max_trials = self.max_trials
>       assert algo.algorithm.max_trials == "BOB"
E       AssertionError: assert 200 == 'BOB'
E        +  where 200 = <orion.algo.asha.ASHA object at 0x7f1d6dad96a0>.max_trials
E        +    where <orion.algo.asha.ASHA object at 0x7f1d6dad96a0> = <orion.core.worker.primary_algo.SpaceTransformAlgoWrapper object at 0x7f1d6d9b0040>.algorithm

src/orion/testing/algo.py:186: AssertionError
```

With the other ordering, we still get the rich output! However, there is an additional warning logged:
```
(...)
>       assert algo.algorithm.max_trials == "BOB"
E       AssertionError: assert 200 == 'BOB'
E        +  where 200 = <orion.algo.asha.ASHA object at 0x7f4dcfa9cb80>.max_trials
E        +    where <orion.algo.asha.ASHA object at 0x7f4dcfa9cb80> = <orion.core.worker.primary_algo.SpaceTransformAlgoWrapper object at 0x7f4dcf9a7370>.algorithm

src/orion/testing/algo.py:186: AssertionError
=============================================================== warnings summary ================================================================
tests/conftest.py:31
  /home/fabrice/Source/orion/tests/conftest.py:31: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: orion.testing
    pytest.register_assert_rewrite("orion.testing")
```

Reply:
Oh, the line above is doing the import already. Ok, both orders are fine. Thanks for verifying!